internal/repositories/github/service.go

```diff
@@ -70,6 +94,65 @@ func NewRepositoryService(
 	}
 }
 
+func (r *repositoryService) CreateRepository(
+	ctx context.Context,
+	client ghclient.GitHubRepoClient,
+	provider *db.Provider,
+	projectID uuid.UUID,
+	repo *pb.UpstreamRepositoryRef,
+) (*pb.Repository, error) {
+	// get information about the repo from GitHub, and ensure it exists
+	githubRepo, err := client.GetRepository(ctx, repo.Owner, repo.Name)
```

Comment:
Slight preference for `.GetOwner()` and `.GetName()`, which will return `""` if `repo == nil`.